File: Dataset/Leetcode/train/118/309.py
 class Solution:
    def XXX(self, nums: int) -> List[List[int]]:
        dicts={}
        def aaaa(nums):
            if nums in dicts:
                return dicts[nums]
            res=[]
            if nums==1:
                res=[1]
            elif nums==2:
                res=[1,1]
            else:
                res=[1]+[sum(aaaa(nums-1)[i-1:i+1]) for i in range(1,nums-1)]+[1]
            dicts[nums]=res
            return res
        return [aaaa(i+1) for i in range(nums)]

File: Dataset/Leetcode/train/125/309.cpp
 class Solution {
public:
    bool XXX(string s) {
        string res ="";
        if(!s.length()) return true;
        for(auto c:s)
        {
            int val = (int)c;
            if(val >= 48 && val <= 57) res+=c;
            else if(val >= 65 && val <= 90) 
            {
                res += (char)(val + 32);
            }
            else if(val >= 97 && val <= 122) res +=c;
        }
        cout<<res<<endl;
        if(!res.length())return true;
        int l = 0; int r =res.length() -1;
        if(res[l] != res[r]) return false;
        while(l <= r)
        {
            if(res[l] != res[r]) return false;
            l++,r--;
        }
        return true;
    }
};

